src/shared/ui/list/list.tsx

```diff
@@ -1,27 +1,46 @@
-// todo delete TextWithLink
+import { HTMLAttributes } from 'react';
+import { type VariantProps, cva } from 'class-variance-authority';
 import classNames from 'classnames/bind';
 import { TextWithLink } from '../text-with-link';
-import { LinkList } from '@/widgets/required/required.types';
+import { ListData } from '@/shared/types';
 
 import styles from './list.module.scss';
 
-export type ListProps = {
-  data: ListData;
-  marked?: boolean;
-};
-
-export type ListData = (string | LinkList)[] | [] | undefined;
+type ListProps = Pick<HTMLAttributes<HTMLElement>, 'className'>
+  & VariantProps<typeof listVariants>
+  & Record<'data', ListData>;
```

Comment:
I'd use `{ data: ListData };` instead of `Record<'data', ListData>` to make it more clear

Reply:
Fixed